src/components/Input/Input.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useCallback } from 'react';
+import debounce from 'lodash.debounce';
+
+type Props = {
+  query: string;
+  setQuery: (value: string) => void;
+  setAppliedQuery: (value: string) => void;
+  setIsFocused: (value: boolean) => void;
+  delay: number;
+};
+
+export const Input: React.FC<Props> = ({
+  query,
+  setQuery,
+  setAppliedQuery,
+  setIsFocused,
+  delay,
+}) => {
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
let's try fix eslint warning and add deps for `useCallback`

Reply:
Sorry, I don't understand "1. hide suggestions on text change". It seems to me I've already implemented it.